src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs

```diff
@@ -47,10 +49,11 @@ public OtlpLogRecordTransformer(SdkLimitOptions sdkLimitOptions, ExperimentalOpt
             var otlpLogRecord = this.ToOtlpLog(logRecord);
             if (otlpLogRecord != null)
             {
-                if (!logsByCategory.TryGetValue(logRecord.CategoryName, out var scopeLogs))
+                var scopeName = logRecord.CategoryName ?? logRecord.Logger?.Name ?? DefaultScopeName;
```

Comment:
Could we just update the sdk to set the logger name to categoryname in `OpenTelemetryLogger`  and use just the logger name here? cc: @CodeBlanch

Reply:
We could. Right now for ILogger path we set [a Logger](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8b54074c7a392b0e295fa7aa0f1a3ab1cb28025a/src/OpenTelemetry/Logs/ILogger/OpenTelemetryLogger.cs#L100) which is named [OpenTelemetry w/ sdk version](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8b54074c7a392b0e295fa7aa0f1a3ab1cb28025a/src/OpenTelemetry/Logs/ILogger/OpenTelemetryLogger.cs#L252). But we could modify the [ctor](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8b54074c7a392b0e295fa7aa0f1a3ab1cb28025a/src/OpenTelemetry/Logs/ILogger/OpenTelemetryLogger.cs#L40) to build a Logger from categoryName.

My only concern would be what to do if `LogRecord.CategoryName` is changed (like inside a processor). It [has a public setter](https://github.com/open-telemetry/opentelemetry-dotnet/blob/8b54074c7a392b0e295fa7aa0f1a3ab1cb28025a/src/OpenTelemetry/Logs/LogRecord.cs#L165). Would we generate a new Logger? Could be an unexpected perf hit to do that.